core/store/doc.go

```diff
@@ -1,3 +1,5 @@
 // Package store provides a basic API for modules to interact with kv-stores
 // independently of any implementation of that functionality.
+// Additionally, it provides a set of interfaces for store implementations to
```

Comment:
I tried to move it, but I eventually didn't like that we ended up with two store packages, and this one imprting to the server one.
I'd rather extend the scope of this package instead, tbh. What do you think @aaronc?